File: mini_browser.js
// mini_browser.js — JavaScript версия (Electron)

const { app, BrowserWindow, globalShortcut } = require('electron');
const path = require('path');
const url = require('url');

let mainWindow;

function createWindow() {
    // Парсим аргументы командной строки
    const args = process.argv.slice(2);
    let targetUrl = 'https://example.com';
    let fullscreen = true;

    for (let i = 0; i < args.length; i++) {
        if (args[i] === '--url') targetUrl = args[++i];
        else if (args[i] === '--no-fullscreen') fullscreen = false;
    }

    mainWindow = new BrowserWindow({
        width: 800,
        height: 600,
        fullscreen: fullscreen,
        frame: false,
        autoHideMenuBar: true,
        webPreferences: {
            nodeIntegration: false,
            contextIsolation: true,
            preload: path.join(__dirname, 'preload.js')
        }
    });

    mainWindow.loadURL(targetUrl);

    // Блокировка выхода
    mainWindow.on('closed', () => {
        mainWindow = null;
    });

    // Регистрация горячих клавиш для выхода
    globalShortcut.register('Escape', () => {
        app.quit();
    });
    globalShortcut.register('CommandOrControl+Shift+Q', () => {
        app.quit();
    });

    console.log(`🌐 Mini-Browser (JavaScript / Electron)`);
    console.log(`Открыт URL: ${targetUrl}`);
    console.log('Для выхода нажмите Esc или Ctrl+Shift+Q');
}

app.whenReady().then(() => {
    createWindow();
});

app.on('window-all-closed', () => {
    if (process.platform !== 'darwin') {
        app.quit();
    }
});

app.on('activate', () => {
    if (BrowserWindow.getAllWindows().length === 0) {
        createWindow();
    }
});

// preload.js (содержимое файла)
// window.addEventListener('DOMContentLoaded', () => {
//     console.log('Kiosk mode active');
// });

File: mini_browser.rs
// mini_browser.rs — Rust версия

use webview::*;
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    let mut url = "https://example.com".to_string();
    let mut fullscreen = true;

    let mut i = 1;
    while i < args.len() {
        match args[i].as_str() {
            "--url" => {
                url = args[i + 1].clone();
                i += 2;
            }
            "--no-fullscreen" => {
                fullscreen = false;
                i += 1;
            }
            _ => i += 1,
        }
    }

    println!("🌐 Mini-Browser (Rust)");
    println!("Открыт URL: {}", url);
    println!("Для выхода нажмите Esc или Ctrl+Shift+Q");

    let mut webview = webview::builder()
        .title("Mini-Browser (Kiosk)")
        .content(Content::Url(&url))
        .size(800, 600)
        .resizable(true)
        .debug(false)
        .user_data(())
        .invoke_handler(|webview, arg| {
            Ok(())
        });

    if fullscreen {
        // Полноэкранный режим зависит от платформы
        // В webview crate это делается через нативные методы
        // Для упрощения используем window.set_fullscreen
        // webview.window().set_fullscreen(true);
    }

    webview.run().unwrap();
}
